Fix warning

@mgautierfr Can you please fix the following warnings:
```
[32/120] Compiling C++ object 'src/25a6634@@zim@sha/archive.cpp.o'.
../src/archive.cpp: In member function ‘std::set<unsigned int> zim::Archive::getIllustrationSizes() const’:
../src/archive.cpp:160:14: warning: variable ‘r’ set but not used [-Wunused-but-set-variable]
  160 |         auto r = findFavicon(*m_impl);
      |              ^
[38/120] Compiling C++ object 'src/25a6634@@zim@sha/tools.cpp.o'.
../src/tools.cpp: In function ‘unsigned int zim::parseIllustrationPathToSize(const string&)’:
../src/tools.cpp:112:14: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::__cxx11::basic_string<char>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
  112 |      && nEnd == s.size() && !isspace(s[nw]) && !isspace(s[nh]) && w == h && w >= 0) {
      |         ~~~~~^~~~~~~~~~~
[120/120] Linking target test/suggestion.
```

_Originally posted by @kelson42 in https://github.com/openzim/libzim/issues/605#issuecomment-896673982_

--- a/src/archive.cpp
+++ b/src/archive.cpp
@@ -157,7 +157,8 @@ namespace zim
     }
     if (ret.find(48) == ret.end()) {
       try {
-        auto r = findFavicon(*m_impl);
+        // raise a exception if we cannot find the (old format) favicon.
+        findFavicon(*m_impl);
         ret.insert(48);
       } catch(EntryNotFound&) {}
     }

--- a/src/tools.cpp
+++ b/src/tools.cpp
@@ -109,7 +109,7 @@ unsigned int zim::parseIllustrationPathToSize(const std::string& s)
   int nw(0), nh(0), nEnd(0);
   long int w(-1), h(-1);
   if ( sscanf(s.c_str(), "Illustration_%n%ldx%n%ld@1%n)", &nw, &w, &nh, &h, &nEnd) == 2
-     && nEnd == s.size() && !isspace(s[nw]) && !isspace(s[nh]) && w == h && w >= 0) {
+     && (size_t)nEnd == s.size() && !isspace(s[nw]) && !isspace(s[nh]) && w == h && w >= 0) {
     return (unsigned int)w;
   }
   throw std::runtime_error("");

